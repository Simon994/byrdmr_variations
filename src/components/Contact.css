/* Contact form  */

.contact-container {
  background: #f5f1ed;
  height: 100vh;
}

.social-container {
  height: 18px;
  width: 90px;
  margin-left: 18px;
  position: relative;
  top: 80px;
  opacity: 0.2;
  display: flex;
  justify-content: space-between;
}

.social-logo {
  height: 100%;
}

.social-logo:hover {
  cursor: pointer;
}

.form-container {
  height: 85vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

form {
  width: 658px;
}

form label,
form input,
textarea {
  display: block;
  width: 100%;
}

form label {
  font-size: 14px;
  margin-bottom: 7px;
  margin-left: 3px;
  letter-spacing: 1px;
}

form input,
textarea {
  font-size: 20px;
  font-family:
    'ProximaNova-Regular',
    'Helvetica Neue',
    sans-serif;
  border: none;
  border-bottom: 1px solid rgba(0, 0, 0, 0.3);
  background: inherit;
  margin-bottom: 25px;
  resize: none;
}

textarea {
  padding-bottom: 60px;
}

.input-regular {
  padding-bottom: 20px;
}

form button {
  background: rgb(238, 95, 32);
  border: none;
  cursor: pointer;
  color: white;
  width: 223px;
  height: 58px;
  font-size: 20px;
  font-family:
    'ProximaNova-Semibold',
    'Helvetica Neue',
    sans-serif;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

form button:hover {
  background: #434343;
  -webkit-box-shadow: 0 5px 40px -10px rgba(0, 0, 0, 0.57);
  -moz-box-shadow: 0 5px 40px -10px rgba(0, 0, 0, 0.57);
  box-shadow: 5px 40px -10px rgba(0, 0, 0, 0.57);
  transition: all 0.4s ease 0s;
}

.arrow {
  width: 17px;
}

@media screen and (max-width: 768px) {
  form {
    width: 647px;
  }
}

@media screen and (max-width: 375px) {
  form {
    width: 322px;
  }
}
