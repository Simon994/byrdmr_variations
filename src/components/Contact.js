import { Component } from 'react' 

import './Contact.css'
import DropdownMenu from './common/DropdownMenu'
import { postMessage } from '../../src/lib/api'

class Contact extends Component {

  state = {
    formData: {
      fullName: '',
      email: '',
      message: ''
    },
    formError: false
  }

  handleChange = (e) => {
    const formData = {
      ...this.state.formData,
      [e.target.name]: e.target.value
    }

    this.setState({
      formData
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    try {
      await postMessage(this.state.formData)

      this.setState({
        formData: {
          fullName: '',
          email: '',
          message: ''
        }
      })

    } catch (err) {
      this.setState({ formError: true })
    }
  }


  render () {
    const {fullName, email, message} = this.state.formData

    return (
      <div className='contact-container'>
        <DropdownMenu/>
        <div className='form-container'>
          <form onSubmit={this.handleSubmit}>
            <label className='form-label'>FULL NAME</label>
            <input placeholder='Enter your full name'
              onChange={this.handleChange}
              value={fullName}
              name='fullName'
              className='input-regular'
            />
            <label className='form-label'>EMAIL</label>
            <input placeholder='Enter your email'
              onChange={this.handleChange}
              value={email}
              name='email'
              className='input-regular'
            />
            <label className='form-label'>MESSAGE</label>
            <textarea placeholder='What are your plans?'
              onChange={this.handleChange}
              value={message}
              name='message'
              className='input-large'
            />
            <button type='submit'>
              <div>
                Send
                Message
              </div>
              <div id='arrow'>
                &rarr;
              </div>
            </button>
          </form>
        </div>
      </div>
    )
  }
}

export default Contact