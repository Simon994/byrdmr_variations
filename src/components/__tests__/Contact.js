import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Contact from '../Contact'

describe('The Contact component', () => {
  it('renders as expected', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Contact/>
      </BrowserRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})