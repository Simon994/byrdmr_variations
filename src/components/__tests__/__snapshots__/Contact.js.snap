// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Contact component renders as expected 1`] = `
<div
  className="contact-container"
>
  <div
    className="social-container"
  >
    <img
      alt="facebook"
      className="social-logo"
      src="fb-logo.png"
    />
    <img
      alt="instagram"
      className="social-logo"
      src="insta-logo.png"
    />
    <img
      alt="twitter"
      className="social-logo"
      src="twitter-logo.png"
    />
  </div>
  <div
    className="outer-menu-container"
  >
    <img
      alt="ByrdMR logo"
      className="byrd-logo"
      src="Logo.png"
    />
    <div
      className="inner-menu-container"
    >
      <button
        className="menu-trigger"
        onClick={[Function]}
      >
        <img
          alt="Menu button icon"
          src="menu.png"
        />
      </button>
      <nav
        className="menu inactive"
      >
        <ul>
          <li>
            <a
              aria-current={null}
              className="nav-link"
              href="/store"
              onClick={[Function]}
            >
              STORE
            </a>
          </li>
          <li>
            <a
              aria-current={null}
              className="nav-link"
              href="/tutorials"
              onClick={[Function]}
            >
              TUTORIALS
            </a>
          </li>
        </ul>
        <ul>
          <li>
            <a
              aria-current={null}
              className="nav-link"
              href="/blog"
              onClick={[Function]}
            >
              BLOG
            </a>
          </li>
          <li>
            <a
              aria-current={null}
              className="nav-link"
              href="/contact"
              onClick={[Function]}
            >
              CONTACT
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    className="form-container"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="form-label"
      >
        FULL NAME
      </label>
      <input
        className="input-regular"
        name="fullName"
        onChange={[Function]}
        placeholder="Enter your full name"
        value=""
      />
      <label
        className="form-label"
      >
        EMAIL
      </label>
      <input
        className="input-regular"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        value=""
      />
      <label
        className="form-label"
      >
        MESSAGE
      </label>
      <textarea
        className="input-large"
        name="message"
        onChange={[Function]}
        placeholder="What are your plans?"
        value=""
      />
      <button
        type="submit"
      >
        <div>
          Send Message
        </div>
        <img
          alt="arrow"
          className="arrow"
          src="button-arrow.png"
        />
      </button>
    </form>
  </div>
</div>
`;
